lib/classes/Variables.test.js

```diff
@@ -341,506 +363,380 @@ module.exports = {
-
-      serverless.variables.loadVariableSyntax();
-
-      getValueFromSourceStub.resolves('prod');
-      populateVariableStub.resolves('my stage is prod');
-
-      return serverless.variables.populateProperty(property).then(newProperty => {
-        expect(getValueFromSourceStub.called).to.be.true;
-        expect(populateVariableStub.called).to.be.true;
-        expect(newProperty).to.equal('my stage is prod');
-
-        return BbPromise.resolve();
-      });
-    });
-
-    it('should NOT call populateObject if variable value is a circular object', () => {
-      serverless.variables.options = {
-        stage: 'prod',
-      };
-      const property = '${opt:stage}';
-      const variableValue = {
-        stage: '${opt:stage}',
-      };
-      const variableValuePopulated = {
-        stage: 'prod',
-      };
-
-      serverless.variables.cache['opt:stage'] = variableValuePopulated;
-
-      serverless.variables.loadVariableSyntax();
-
-      populateObjectStub.resolves(variableValuePopulated);
-      getValueFromSourceStub.resolves(variableValue);
-      populateVariableStub.resolves(variableValuePopulated);
-
-      return serverless.variables.populateProperty(property).then(newProperty => {
-        expect(populateObjectStub.called).to.equal(false);
-        expect(getValueFromSourceStub.called).to.equal(true);
-        expect(populateVariableStub.called).to.equal(true);
-        expect(newProperty).to.deep.equal(variableValuePopulated);
-
-        return BbPromise.resolve();
-      });
+      serverless.variables.options = { stage: 'prod' };
+      return serverless.variables.populateProperty(property)
+        .should.eventually.eql('my stage is prod');
     });
 
     it('should warn if an SSM parameter does not exist', () => {
-      const awsProvider = new AwsProvider(serverless, { stage: 'prod', region: 'us-west-2' });
-      const param = '/some/path/to/invalidparam';
-      const property = `\${ssm:${param}}`;
-      const error = new Error(`Parameter ${param} not found.`);
-
-      serverless.variables.options = {
+      const options = {
         stage: 'prod',
         region: 'us-east-1',
       };
-      serverless.variables.loadVariableSyntax();
-
-      serverless.variables.getValueFromSource.restore();
-      serverless.variables.populateVariable.restore();
+      serverless.variables.options = options;
+      const awsProvider = new AwsProvider(serverless, options);
+      const param = '/some/path/to/invalidparam';
+      const property = `\${ssm:${param}}`;
+      const error = new Error(`Parameter ${param} not found.`, 123);
       const requestStub = sinon.stub(awsProvider, 'request', () => BbPromise.reject(error));
       const warnIfNotFoundSpy = sinon.spy(serverless.variables, 'warnIfNotFound');
-
-      return expect(serverless.variables.populateProperty(property)
-        .then(newProperty => {
+      return serverless.variables.populateProperty(property)
+        .should.become(undefined)
```

Comment:
we use `chai-as-promised`, so you can model this with expect instead of introducing the should semantics.
```
return expect(serverless.variables.populateProperty(property)).to.eventually.be.undefined
.then(() => ...
```

Reply:
chai-as-promised provides these should semantics to promises as well.  this seemed much cleaner.

Is there a reason not to use should here?
Is there a standard to avoid it?

Reply:
@erikerikson I do not have a special opinion here. I just was curious why you switched to should here :)

Reply:
👍

Thanks for explaining.  I'm happy to go either way.  I stumbled into should elsewhere and it has sometimes made tests more legible.

I've actually avoided it on the sister branch to this one where I'm trying to debug the build/tests (#4716).  I'd be happy to import that work here but mildly prefer should.  Just say the word and I'll do that but otherwise I'll assume it's fine.

Reply:
Go for should then :)